README.md

```diff
@@ -1,32 +1,122 @@
 Lasso
 =====
 
-Core controller framework used by [Wrangler](http://github.com/rancher/wrangler) and [Norman](http://github.com/rancher/norman).
+# Contents
+- [Lasso](#lasso)
+- [Contents](#contents)
+- [Overview](#overview)
+- [Structure of this Document](#structure-of-this-document)
+- [Kubernetes Controllers](#kubernetes-controllers)
+- [Lasso's Value Proposition](#lassos-value-proposition)
+	- [Lasso Controller](#lasso-controller)
+	- [Lasso Shared Controller](#lasso-shared-controller)
+	- [Lasso Client](#lasso-client)
+	- [Lasso Cache](#lasso-cache)
+	- [Lasso Shared Cache Factory](#lasso-shared-cache-factory)
+	- [Lasso Shared Client Factory](#lasso-shared-client-factory)
+	- [Lasso Shared Controller Factory](#lasso-shared-controller-factory)
+	- [Deferred Controller Function](#deferred-controller-function)
+- [How to use Lasso](#how-to-use-lasso)
+- [Credits and Resources](#credits-and-resources)
 
-This framework is a bit raw, but can be used directly. The intention is try to standardize some boilerplate
-of building controllers.
+# Overview
+Lasso is a controller framework.
 
-### Basic components
+Lasso has some functional parity with the controller-runtime project and could be used in place of controller-runtime.
 
-#### Package `cache`
+Lasso is low level. If you want to manage your own controllers then you can use it. If you are looking for more convenience and functionality for creating your own resource types, we recommend checking out [Wrangler](https://github.com/rancher/wrangler).
 
-This is a generic form of the generated SharedIndexFactory.
+Core controller framework used by [Wrangler](https://github.com/rancher/wrangler) and [Norman](https://github.com/rancher/norman).
 
-#### Package `client`
+# Structure of this Document
+This document was written with two audiences in mind: someone who is new to Kubernetes controllers and someone who is not new to Kubernetes controllers but is new to lasso.
 
-Generic form of the generated clientsets.
+If you fall into the category of "Someone who is not new to Kubernetes controllers" you may skip to [Lasso's Value Proposition](#lassos-value-proposition).
 
-#### Package `controller`
+If you are new to Kubernetes controllers it is recommended you do not skip the any sections.
 
-The `Controller` is a simple approach that wraps the standard pattern of worker goroutines and workqueue.  The package
-depends on the `cache` and `client` removing the need for generated clientsets, informers, and listers.
+# Kubernetes Controllers
+_Note:_ This will be a very brief crash course. We encourage you to look up other resources regarding the topic, there are many!
```

Comment:
Optional: offer few links to high-quality material.

Rationale: beginners struggle when there is abundance of low quality material.

Personally https://github.com/aiyengar2/k8s-docs/blob/main/docs/controllers/00_introduction.md and the https://web.microsoftstream.com/video/967a4311-fb11-4637-90c9-a4be4c180dad series of videos helped - but I was not starting completely from scratch.

Reply:
The second link is not open to everyone. I can link the sample controller project instead of that one.